app/test/shared/handlers_test_utils.dart

```diff
@@ -5,42 +5,18 @@
 import 'dart:async';
 import 'dart:convert';
 
-import 'package:logging/logging.dart';
+import 'package:api_builder/_client_utils.dart' show RequestException;
 import 'package:shelf/shelf.dart' as shelf;
 import 'package:test/test.dart';
 
-import 'package:pub_dartlang_org/frontend/handlers.dart';
-import 'package:pub_dartlang_org/shared/handler_helpers.dart';
-import 'package:pub_dartlang_org/shared/urls.dart';
-
-/// Send an API request for [method] and [uri] with an optional [jsonBody].
-/// The request will use the wrapped frontend handler and will not use real HTTP
-/// requests.
-Future<shelf.Response> httpRequest(
-  String method,
-  dynamic uri, {
-  String authToken,
-  Map<String, dynamic> jsonBody,
-}) async {
-  if ('$uri'.startsWith('/')) {
-    uri = '$siteRoot$uri';
-  }
-  final request = shelf.Request(
-    method,
-    uri is Uri ? uri : Uri.parse(uri.toString()),
-    headers: {
-      if (authToken != null) 'authorization': 'bearer $authToken',
-      if (jsonBody != null) 'content-type': 'application/json; charset=utf-8',
-      if (jsonBody != null) 'accept': 'application/json',
-    },
-    body: jsonBody == null ? null : json.encode(jsonBody),
-  );
-  final handler = wrapHandler(
-    Logger.detached('test'),
-    createAppHandler(null),
-    sanitize: true,
-  );
-  return await handler(request);
+Future expectApiException(
+    Future future, int status, String code, String message) async {
```

Comment:
I think it would be fine if `status`, `code` and `message` was optional.

I don't think we should make tests covering `message`, that's too fragile... If we want we can pass a `Matcher` that message should match... Or accept a message substring..

Reply:
Updated: the optional message will be matched with `contains(message)`.